src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -269,31 +254,41 @@ private void initCustomStyleOptions() {
   private void initSeedField() {
     seedTextField.setText(String.valueOf(generatorPrefs.getSeed()));
     generatorPrefs.seedProperty().bind(seedTextField.textProperty());
-    seedTextField.disableProperty()
-                 .bind(previousMapName.textProperty()
-                                      .isNotEmpty()
-                                      .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                      .or(commandLineArgsText.textProperty().isNotEmpty())
-                                      .or(fixedSeedCheckBox.selectedProperty().not()));
-    seedRerollButton.disableProperty()
-                    .bind(previousMapName.textProperty()
-                                         .isNotEmpty()
-                                         .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                         .or(commandLineArgsText.textProperty().isNotEmpty())
-                                         .or(fixedSeedCheckBox.selectedProperty().not()));
-  }
-
-  private void bindCustomStyleDisabledProperty(Node... nodes) {
+    seedTextField.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+    seedRerollButton.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+  }
+
+  private BooleanBinding getSharedBooleanBinding() {
+    return previousMapName.textProperty().isNotEmpty()
+                          .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
+                          .or(commandLineArgsText.textProperty().isNotEmpty());
+  }
+
+  private BooleanBinding getExtendedFixedSeedNotBooleanBinding() {
+    return getSharedBooleanBinding()
+        .or(fixedSeedCheckBox.selectedProperty().not());
+  }
+
+  private BooleanBinding getExtendedCustomStyleNotBooleanBinding() {
+    return getSharedBooleanBinding()
+        .or(customStyleCheckBox.selectedProperty().not());
+  }
+
+  private BooleanBinding getExtendedSliderBooleanBinding(ComboBox<String> comboBox) {
+    return getExtendedCustomStyleNotBooleanBinding()
+        .or(comboBox.valueProperty().isEqualTo(MapGeneratorService.GENERATOR_RANDOM_OPTION));
+  }
```

Comment:
This function is only used in one place so I think it should just be inlined

Reply:
This is used twice. It is cause of the whole mess. 2 separate sliders set..

Reply:
It is only called in the function bindCustomStyleDisabledPropertySlider

Reply:
yeah, but the function is called twice and each time with dif. comboBox - resource slider(dedicated combobox) vs props/reclaim slider (+ dedicated combobox)